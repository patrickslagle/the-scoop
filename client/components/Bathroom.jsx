import React from 'react';
import ReactMapboxGl, { Marker } from "react-mapbox-gl";
import styled from 'styled-components';


const Bathroom = (props) => {
  const Mark = styled.div`
  background-color: #e74c3c;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  border: 4px solid #eaa29b;
`;
  return (
    <div>
      <Marker
        coordinates={props.coordinate}
        anchor="bottom"
      >
        <Mark />
      </Marker>
    </div>
  );
};

export default Bathroom;
