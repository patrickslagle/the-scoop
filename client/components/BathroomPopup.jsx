import React from 'react';
import ReactMapboxGl, { Popup } from "react-mapbox-gl";


const BathroomPopup = props => {
  return (
  <div>
    <Popup className="popup"
      coordinates={props.coordinate}
      // onClick={() => this.addLike({test: 'testing'})}
      anchor="bottom"
      offset={{
        'bottom-left': [0, 0],  'bottom': [0, 0], 'bottom-right': [0, 0]
      }}>
    <h1>Add Bathroom</h1>
    <form onSubmit={props.submitHandler}> 
      <input type="text"
        placeholder="Bathroom Location"
        value={props.bathroomLocation} 
        onChange={props.handleChange} 
      /><br/>
      <input type="text"
        placeholder="Bathroom URL Pic"
        value={props.bathroomPic} 
        onChange={props.handleChange}  
      /><br/>
      <input type="text"
        placeholder="Poop Review"
        value={props.review} 
        onChange={props.handleChange}  
      /><br/>
      <input type="submit" 
        value="Submit"
      />
    </form>
  </Popup>
  </div>
)};

export default BathroomPopup; 

