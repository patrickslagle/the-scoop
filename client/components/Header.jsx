import React from 'react';

const Header = () => (
  <div>
    <header>
      <h1>The Poop Scoop ðŸ’©</h1>
    </header>
    <div className="header-pic">
      <img
        src="https://outsider.ie/wp-content/uploads/2017/08/ALT-HEADER-3401601513_80ea646cd9_o-1980x450.jpg" 
        alt="get the scoop on where to poop"
      />
    </div>
  </div>
);

export default Header;
