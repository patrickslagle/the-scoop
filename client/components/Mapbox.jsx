import React, { Component } from 'react';
import ReactMapboxGl from 'react-mapbox-gl';
import axios from 'axios';
import Bathroom from './Bathroom.jsx';
import BathroomPopup from './BathroomPopup.jsx';
// AWESOME map examples below
// http://alex3165.github.io/react-mapbox-gl/demos
// https://www.mapbox.com/mapbox-gl-js/api/

const TOKEN = 'pk.eyJ1Ijoic2xhZ2xlYmFnZWwxNyIsImEiOiJjamw1ZHgxc3MwcmlsM3FxbHIzeml1bDFlIn0.MgcX0bSt1txgmOR-HOGlqA';

const Map = ReactMapboxGl({
  accessToken: TOKEN
});

class Mapbox extends Component{
  constructor(props){
    super(props)
    this.state = {
      viewport: {
        latitude: 37.785164,
        longitude: -100,
        zoom: 2.8,
        bearing: 0,
        pitch: 0,
        width: 500,
        height: 500,
      },
      popup: {
        bathroomLocation: "",
        review: "",
        bathroomPic: "",
        coordinate: null
      },
      bathrooms: []
    }
    this.submitHandler = this.submitHandler.bind(this)
    this.handleChange = this.handleChange.bind(this)
    this.bathroomPopup = this.bathroomPopup.bind(this)
  }

  componentDidMount(){
    axios.get('/loadBathrooms').then(bathrooms => {
        console.log('loaded bathrooms ', bathrooms.data);
        this.setState({bathrooms: bathrooms.data})
      })
  }
  
  bathroomPopup(map, evt){
    const coordinate = [evt.lngLat.lng, evt.lngLat.lat]
    const newBathroom = this.state.popup
    this.state.popup.coordinate = coordinate
    this.setState({popup: newBathroom})    
    console.log(this.state.popup)
  }

  //submitting bathroom info
  submitHandler(event){
    event.preventDefault();
    const newBathroom = this.state.popup
    const bathrooms = this.state.bathrooms
    axios.post('/addBathroom', {newBathroom}).then(response => {
      console.log('in the axios post, this is response.data', response.data)
      // this.setState({bathrooms: bathrooms.data})
      bathrooms.push(response.data);
      this.setState({bathrooms: bathrooms})
    })
  }

  //if text is entered into a popup
  handleChange(event){
    if (event.target.placeholder === "Bathroom Location"){
      const tempPopup = this.state.popup
      tempPopup.bathroomLocation = event.target.value; 
      this.setState({popup: tempPopup})
      console.log(event.target.value)
    }
    else if (event.target.placeholder === "Bathroom URL Pic"){
      const tempPopup = this.state.popup
      tempPopup.bathroomPic = event.target.value; 
      this.setState({popup: tempPopup})
      console.log(event.target.value)

    }
    else if (event.target.placeholder === "Poop Review"){
      const tempPopup = this.state.popup
      tempPopup.review = event.target.value; 
      this.setState({popup: tempPopup})
      console.log(event.target.value)
    }
  }


  render(){
    const {viewport} = this.state;
    const bathrooms = this.state.bathrooms.map(bathroom => {
        return <Bathroom key={bathroom._id} coordinate={bathroom.coordinate}/>
    })
    let newPopup; 
    if (this.state.popup.coordinate) {
      newPopup = <BathroomPopup 
                    coordinate={this.state.popup.coordinate}
                    submitHandler={this.submitHandler}
                    handleChange={this.handleChange}
                    bathroomLocation={this.state.popup.bathroomLocation}
                    review={this.state.popup.review}
                    bathroomPic={this.state.popup.bathroomPic}
                    />
    }
    return(
      <div className="map-container">
        <Map className="map-container"
          onClick={this.bathroomPopup}
    
          style="mapbox://styles/mapbox/streets-v9"
          containerStyle={{
            height: "60vh",
            width: "92vw",
            margin: "2.8%",
          }}
          >
          {newPopup}
          {bathrooms}
        </Map>
      </div>
    )
  }
}

export default Mapbox