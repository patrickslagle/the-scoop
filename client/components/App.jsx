import React, { Component } from 'react'
import axios from 'axios'
import Mapbox from './Mapbox.jsx'
import Header from './Header.jsx'

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
        test: 'test',
    }
  }

  render(){
    return(
      <div className="app-container">
        <Header/>
        <Mapbox/>
      </div>
    )
  }
}

export default App; 