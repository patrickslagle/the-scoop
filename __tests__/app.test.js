import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import App from '../client/components/App.jsx';
import Bathroom from '../client/components/Bathroom.jsx';
import BathroomPopup from '../client/components/BathroomPopup.jsx';
import Header from '../client/components/Header.jsx';
import Mapbox from '../client/components/Mapbox.jsx';

configure({ adapter: new Adapter() });

// solve jest error when trying to import react-mapbox-gl
jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
  Map: () => ({}),
}));

test('App component renders properly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

test('Bathroom component renders properly', () => {
  const wrapper = shallow(<Bathroom />);
  expect(wrapper).toMatchSnapshot();
});
