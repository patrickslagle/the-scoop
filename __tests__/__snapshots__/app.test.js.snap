// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders properly 1`] = `
<div
  className="app-container"
>
  <Header />
  <Mapbox />
</div>
`;

exports[`Bathroom component renders properly 1`] = `
<div>
  <Marker
    anchor="bottom"
  >
    <styled.div />
  </Marker>
</div>
`;
